src/evaluator/function/comparisons.cpp

```diff
@@ -0,0 +1,78 @@
+#include "../error.h"
+#include "../function.h"
+
+namespace evaluator {
+
+using ast::Element;
+
+std::shared_ptr<Element> EqualFunction::call(CallFrame frame) const {
+    if (frame.arguments.size() != 2) {
+        throw EvaluationError(
+            "`equal` expects 2 arguments, received " +
+                std::to_string(frame.arguments.size()),
+            frame.call_site
+        );
+    }
+
+    auto a_element = frame.arguments[0];
+    auto b_element = frame.arguments[1];
+
+    bool result;
+
+    if (auto a_real = std::dynamic_pointer_cast<ast::Real>(a_element)) {
+        if (auto b_real = std::dynamic_pointer_cast<ast::Real>(b_element)) {
+            double a = a_real->value;
+            double b = b_real->value;
+
+            result = a == b;
+        } else {
+            throw EvaluationError(
+                "`equal` expects that arguments have the same type",
+                frame.call_site
+            );
+        }
+    } else if (auto a_int = std::dynamic_pointer_cast<ast::Integer>(a_element)) {
+        if (auto b_int = std::dynamic_pointer_cast<ast::Integer>(b_element)) {
+            int a = a_int->value;
+            int b = b_int->value;
+
+            result = a == b;
+        } else {
+            throw EvaluationError(
+                "`equal` expects that arguments have the same type",
+                frame.call_site
+            );
+        }
+    } else if (auto a_bool = std::dynamic_pointer_cast<ast::Boolean>(a_element)) {
+        if (auto b_bool = std::dynamic_pointer_cast<ast::Boolean>(b_element)) {
+            bool a = a_bool->value;
+            bool b = b_bool->value;
+
+            result = a == b;
+        } else {
+            throw EvaluationError(
+                "`equal` expects that arguments have the same type",
+                frame.call_site
+            );
+        }
+    } else {
+        throw EvaluationError(
+            "`equal` expects that arguments can have only the following types: "
+            "Integer, Real, Boolean",
+            frame.call_site
+        );
+    }
+
+    auto result_element =
+        std::make_shared<ast::Boolean>(ast::Boolean(result, this->span));
+
+    return result_element;
+}
+
+std::string_view EqualFunction::name() const { return "equal"; }
+
+void EqualFunction::display_parameters(std::ostream& stream) const {
+    stream << "element element";
```

Comment:
```suggestion
    stream << "a b";
```

не могут же параметры иметь одинаковое имя